### DISCUSSION TITLE
What are the breaking changes in 9.0.0?

### DISCUSSION THREAD

**Discussion by ClubAssist-PatrickBoyer** [CommunityToolkit/Maui#1881] [Q&A] [answered]
What are the breaking changes mentioned in the release notes for 9.0.0?

I acknowledge that compile-time we will know pretty quickly. Through reflection, we could experience crashes at runtime. Having a better understanding of the breaking changes will give us confidence of upgrading the library to the latest version.

I searched for "9" and "Breaking Changes" across the existing discussions already and didn't find anything. Any information would be great appreciated. 

Thanks!
Patrick

**Comment by bijington** [comment]
@ClubAssist-PatrickBoyer I've been meaning to update the release notes so thanks for the reminder. I have added some details here: https://github.com/CommunityToolkit/Maui/releases/tag/9.0.0. The breaking changes are unlikely to impact many developers but they do change some public APIs so they are technically breaking.

I understand that using reflection at runtime could introduce issues and I truly hope that developers are not using reflection in their use with the toolkit. Are you and if so why? Is there something that we haven't exposed?

**Reply by TheCodeTraveler** [reply, marked as answer]
Thanks Shaun!

And for future reference, you can use this quick search for the `breaking change` label to see if any of the PRs with breaking changes impact the features you're using:

https://github.com/CommunityToolkit/Maui/pulls?q=label%3A%22breaking+change%22+

**Reply by ClubAssist-PatrickBoyer** [reply]
To close the loop on this discussion, I confirm that we do not use reflection with the Community Toolkit but we have been burned in the past by linker issues traced back to reflection. So, I was pointing it out of an abundance of caution and in the spirit of thoroughness.

I appreciate Brandon also mentioning the breaking change tag.

Regards,
Patrick

**Comment by CliffAgius** [comment]
Closed as Answered.
